mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -0,0 +1,261 @@
+		//		the the type of sensor and its ID. p.e.: topics "laser_distance_1", "laser_distance_2"...
+
+		if(rangefinder_type == 0)
+			dist_laser_in_pub = dist_nh.advertise<mavros_extras::LaserDistanceSensor>("laser_distance", 10);
+		else if (rangefinder_type == 1)
+			dist_sonar_in_pub = dist_nh.advertise<mavros_extras::SonarDistanceSensor>("sonar_distance", 10);
+		else ROS_ERROR_NAMED("rangefinder", "Invalid rangefinder type! Valid values are 0 (laser) and 1 (ultrasound)!");
+		
+		dist_laser_sub = dist_nh.subscribe("laser_range", 10, &DistanceSensorPlugin::laser_dist_sensor_cb, this);
+		dist_sonar_sub = dist_nh.subscribe("sonar_range", 10, &DistanceSensorPlugin::sonar_dist_sensor_cb, this);
+		
+	}
+
+	const message_map get_rx_handlers() {
+		return {
+			       MESSAGE_HANDLER(MAVLINK_MSG_ID_DISTANCE_SENSOR, &DistanceSensorPlugin::handle_dist_sensor)
+		};
+	}
+
+private:
+	ros::NodeHandle dist_nh;
+	UAS *uas;
+
+	/* -*- variables -*- */
+	std::string frame_id;
+
+	double sonar_fov;
+	double sonar_min_range;
+	double laser_min_range;
+	double sonar_max_range;
+	double laser_max_range;
+
+	int rangefinder_type;
+	int sonar_id;
+	int laser_id;
+	int sonar_orientation;
+	int laser_orientation;
+	int sonar_covariance;
+	int laser_covariance;
+
+	/* -*- publishers -*- */
+	ros::Publisher dist_laser_in_pub;
+	ros::Publisher dist_sonar_in_pub;
+
+	/* -*- subscribers -*- */
+	ros::Subscriber dist_laser_sub;
+	ros::Subscriber dist_sonar_sub;
+
+	/* -*- low-level send -*- */
+	void distance_sensor(uint32_t time_boot_ms,
+				uint32_t min_distance,
+				uint32_t max_distance,
+				uint32_t current_distance,
+				uint8_t type, uint8_t id,
+				uint8_t orientation, uint8_t covariance) {
+		mavlink_message_t msg;
+		mavlink_msg_distance_sensor_pack_chan(UAS_PACK_CHAN(uas), &msg,
+				time_boot_ms,
+				min_distance,
+				max_distance,
+				current_distance,
+				type,
+				id,
+				orientation,
+				covariance);
+		UAS_FCU(uas)->send_message(&msg);
+	}
+
+	/* -*- mid-level helpers -*- */
+
+	/**
+	 * Receive distance sensor data from FCU.
+	 */
+	void handle_dist_sensor(const mavlink_message_t *msg, uint8_t sysid, uint8_t compid) {
+		mavlink_distance_sensor_t dist_sen;
+		mavlink_msg_distance_sensor_decode(msg, &dist_sen);
+
+		std_msgs::Header header;
+		header.stamp = ros::Time::now();
+		header.frame_id = frame_id;
+
+		sensor_msgs::Range sonar;
+		sensor_msgs::LaserScan laser;
+
+		if (dist_sen.type == 0) { // Laser type
+			auto dist_laser_msg = boost::make_shared<mavros_extras::LaserDistanceSensor>();
+
+			dist_laser_msg->header = header;
+			dist_laser_msg->time_boot_ms = dist_sen.time_boot_ms;
+
+			dist_laser_msg->laser.header = header;
+			dist_laser_msg->laser.angle_min = 0.0;
+			dist_laser_msg->laser.angle_max = 0.0;
+			dist_laser_msg->laser.angle_increment = 0.0;
+			dist_laser_msg->laser.time_increment = 0.0;
+			dist_laser_msg->laser.scan_time = 0.0;
+			dist_laser_msg->laser.range_min = laser_min_range/1E2; 
```

Comment:
Yeah you are right, since the value from the FCU has to come in centimeters already.